src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -90,42 +125,128 @@ def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
+            whether need to re-generate configuration of kea-dhcp-server
         """
-        need_refresh = False
-        while self.subscribe_vlan_intf_table.hasData():
-            key, _, _ = self.subscribe_vlan_intf_table.pop()
-            splits = key.split("|")
-            vlan_name = splits[0]
-            ip_address = splits[1].split("/")[0] if len(splits) > 1 else None
-            if vlan_name not in enabled_dhcp_interfaces:
-                continue
-            if ip_address is None or ipaddress.ip_address(ip_address).version != 4:
-                continue
-            need_refresh = True
+        state, _ = self.sel.select(self.select_timeout)
+        if state == swsscommon.Select.TIMEOUT or state != swsscommon.Select.OBJECT:
+            return False
+        need_refresh = self._check_dhcp_server_update(enabled_dhcp_interfaces)
+        need_refresh |= self.db_event_executor.check_vlan_update(enabled_dhcp_interfaces, self.subscribe_vlan_table)
+        need_refresh |= self.db_event_executor.check_vlan_intf_update(enabled_dhcp_interfaces,
+                                                                      self.subscribe_vlan_intf_table)
+        need_refresh |= self._check_dhcp_server_port_update(enabled_dhcp_interfaces)
+        need_refresh |= self._check_dhcp_server_range_update(used_range)
+        need_refresh |= self._check_dhcp_server_option_update(used_options)
+        need_refresh |= self._check_vlan_member_update(enabled_dhcp_interfaces)
         return need_refresh
+
+    def _check_dhcp_server_update(self, enabled_dhcp_interfaces):
+        need_refresh = False
+        while self.subscribe_dhcp_server_table.hasData():
+            key, op, entry = self.subscribe_dhcp_server_table.pop()
+            # If old state is enabled, need refresh
+            if key in enabled_dhcp_interfaces:
+                need_refresh = True
+            elif op == "SET":
+                for field, value in entry:
+                    if field != "state":
+                        continue
+                    # If old state is not consistent with new state, need refresh
+                    if value == "enabled":
+                        need_refresh = True
+            if need_refresh:
+                self.db_event_executor.pop_db_update_event(self.subscribe_dhcp_server_table)
+                return True
+
+        return False
+
+    def _check_dhcp_server_port_update(self, enabled_dhcp_interfaces):
+        while self.subscribe_dhcp_server_port_table.hasData():
+            key, _, _ = self.subscribe_dhcp_server_port_table.pop()
+            dhcp_interface = key.split("|")[0]
+            # If dhcp interface is enabled, need to generate new configuration
+            if dhcp_interface in enabled_dhcp_interfaces:
+                self.db_event_executor.pop_db_update_event(self.subscribe_dhcp_server_port_table)
+                return True
+        return False
+
+    def _check_dhcp_server_range_update(self, used_range):
```

Comment:
enabled_dhcp_interfaces/used_range/used_options can be queried from db, can we query it from db instead pass it to parameter list?

Reply:
These data indicate setting of last time, query data is the newest data